upup/pkg/fi/fitasks/managedfile.go

```diff
@@ -60,13 +63,23 @@ func (e *ManagedFile) Find(c *fi.Context) (*ManagedFile, error) {
 		return nil, err
 	}
 
+	existingDataString := string(existingData)
```

Comment:
This appears to be adding the assumption that a `ManagedFile` is reasonably representable in a `string`. I don't think that's the case: it could be pure binary data.

One can take the SHA1 of a `[]byte`, so converting it to a `string` is not necessary.

If we're going to bake knowledge of hashes into `ManagedFile`, we shouldn't be doing SHA1. It should be SHA256.

Reply:
I honestly did not look at the `HashString` method. The point was to not break any tests. If we want to change the hash function to sha256, I am all for, but as another PR.